src/interpolation/ArborX_InterpMovingLeastSquares.hpp

```diff
@@ -0,0 +1,272 @@
+    static_assert(GeometryTraits::is_point<tgt_point>::value,
+                  "Target points elements must be points");
+    static_assert(dimension == GeometryTraits::dimension_v<tgt_point>,
+                  "Target and source points must have the same dimension");
+
+    num_neighbors =
+        (num_neighbors <= 0)
+            ? Details::polynomialBasisSize<dimension, PolynomialDegree::value>()
+            : num_neighbors;
+
+    int const num_targets = tgt_acc::size(target_points);
+    _source_size = source_points.extent(0);
+    // There must be enough source points
+    ARBORX_ASSERT(num_neighbors <= _source_size);
+
+    // Organize the source points as a tree
+    using src_coord = typename GeometryTraits::coordinate_type<src_point>::type;
+    using box = ExperimentalHyperGeometry::Box<dimension, src_coord>;
+    using bvh = BasicBoundingVolumeHierarchy<
+        MemorySpace, ArborX::Details::PairIndexVolume<box>,
+        ArborX::Details::DefaultIndexableGetter, box>;
+    bvh source_tree(space, source_points);
+
+    // Create the predicates
+    Details::MLSTargetPointsPredicateWrapper<TargetPoints> predicates{
+        target_points, num_neighbors};
+
+    // Query the source
+    Kokkos::View<int *, MemorySpace> indices(
+        "ArborX::MovingLeastSquares::indices", 0);
+    Kokkos::View<int *, MemorySpace> offsets(
+        "ArborX::MovingLeastSquares::offsets", 0);
+    source_tree.query(space, predicates, indices, offsets);
+
+    // Fill in the value indices object so values can be transferred from a 1D
+    // source data to a properly distributed 2D array for each target and do it
+    // for source points
+    auto const source_view = fillValuesIndicesAndGetSourceView(
+        space, indices, offsets, num_targets, num_neighbors, source_points);
+
+    // Compute the Moving Least Squares
+    _coeffs = Kokkos::View<FloatingCalculationType **, MemorySpace>(
+        "ArborX::MovingLeastSquares::coefficients", 0, 0);
+    Details::movingLeastSquaresCoefficients<CRBF, PolynomialDegree>(
+        space, source_view, target_points, _coeffs);
+  }
+
+  template <typename ExecutionSpace, typename SourcePoints,
+            typename TargetPoints, typename CRBF, typename PolynomialDegree>
+  MovingLeastSquares(ExecutionSpace const &space,
+                     SourcePoints const &source_points,
+                     TargetPoints const &target_points, CRBF, PolynomialDegree)
+      : MovingLeastSquares(space, source_points, target_points, 0, CRBF{},
+                           PolynomialDegree{})
+  {}
+
+  template <typename ExecutionSpace, typename SourcePoints,
+            typename TargetPoints>
+  MovingLeastSquares(ExecutionSpace const &space,
+                     SourcePoints const &source_points,
+                     TargetPoints const &target_points)
+      : MovingLeastSquares(space, source_points, target_points,
+                           CRBF::Wendland<2>{}, PolynomialDegree<2>{})
```

Comment:
We need to document default values. Why did you choose these?

Reply:
They were the ones that yielded the best results in my firsts small examples. We could discuss on which default would actually be best.